### ISSUE TITLE
Bulk foreign write in batch-add-sequence

### ISSUE THREAD

**Issue by licjon** [licjon/cl-llama-cpp#108]
## Problem

`batch-add-sequence` (batch.lisp:151) adds tokens by calling `batch-add-token` in a loop:

```lisp
(dotimes (i n)
  (batch-add-token batch (aref tokens i) (the fixnum (+ start-pos i)) seq-id
                   :logits (ecase logits
                             (:last (= i (1- n)))
                             (:all t)
                             ((nil) nil)))))
```

Each iteration of this loop:

1. Calls `batch-add-token` (function call overhead)
2. Runs `%batch-check-overflow` (reads and compares two fixnums)
3. Writes one token, one position, one seq-id, and one logits flag through CFFI

For large prompts (thousands of tokens), the per-token function-call and overflow-check overhead is measurable — especially since all the data is contiguous in the foreign batch struct and could be filled in bulk.

## Proposed change

Write the token, pos, and seq-id arrays directly into the batch's foreign buffers using a single loop over `cffi:mem-aref`, with one overflow check up front instead of per-token:

```lisp
(defun batch-add-sequence (batch tokens seq-id &key (start-pos 0) (logits :last))
  ;; ... validation unchanged ...
  (let* ((n (length tokens))
         (offset (batch-token-count batch))
         (data (%batch-data batch))
         (tok-ptr (getf data '%llama:token))
         (pos-ptr (getf data '%llama:pos))
         (n-seq-ptr (getf data '%llama:n-seq-id))
         (seq-ptr (getf data '%llama:seq-id))
         (logit-ptr (getf data '%llama:logits)))
    (declare (type fixnum n offset start-pos))
    ;; Single overflow check for the whole sequence
    (when (> (+ offset n) (%batch-capacity batch))
      (error 'batch-overflow-error
             :capacity (%batch-capacity batch)
             :token-count (+ offset n)))
    ;; Bulk write
    (dotimes (i n)
      (declare (type fixnum i))
      (let ((idx (+ offset i)))
        (setf (cffi:mem-aref tok-ptr '%llama:token idx) (aref tokens i))
        (setf (cffi:mem-aref pos-ptr '%llama:pos idx) (+ start-pos i))
        (setf (cffi:mem-aref n-seq-ptr :int32 idx) 1)
        (setf (cffi:mem-ref (cffi:mem-aref seq-ptr :pointer idx)
                            '%llama:seq-id)
              seq-id)
        (setf (cffi:mem-aref logit-ptr :int8 idx)
              (ecase logits
                (:last (if (= i (1- n)) 1 0))
                (:all 1)
                ((nil) 0)))))
    (setf (getf (%batch-data batch) '%llama:n-tokens) (+ offset n)))
  nil)
```

The exact struct field access may need adjustment based on the batch plist layout, but the pattern is: one bounds check, one tight loop, direct foreign writes.

## Impact

Eliminates N function calls and N overflow checks per `batch-add-sequence`, replaced by one check and one inlined loop. Most noticeable on large-prompt prefill where N can be 2k–8k tokens.
